app/services/attribute_asserter.rb

```diff
@@ -42,7 +42,7 @@ def build
       add_vot(attrs)
     else
       add_aal(attrs)
-      add_ial(attrs) if authn_request.requested_ial_authn_context || !service_provider.ial.nil?
+      add_ial(attrs) if requested_ial_authn_context || !service_provider.ial.nil?
```

Comment:
since `requested_ial_authn_context` is only used in the case of this conditional, i might pull the whole conditional into a method (and/or update the `.nil?` check to a `present?` check so it can be positive) but it's not a big deal really